sw/device/tests/otbn_mem_scramble_test.c

```diff
@@ -0,0 +1,101 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_base.h"
+#include "sw/device/lib/dif/dif_otbn.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/otbn.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTBN_DECLARE_APP_SYMBOLS(randomness);
+static const otbn_app_t kOtbnAppCfiTest = OTBN_APP_T_INIT(randomness);
+
+const test_config_t kTestConfig;
+
+/**
+ * Check whether the content of an otbn memory match the reference data pointed
+ * by `addr`.
+ *
+ * @param ctx The otbn context object.
+ * @param addr The address to the reference data.
+ * @param mem_size The size to the reference data.
+ * @param read_function Pointer to the function to read the data. It can be
+ * either `dif_otbn_imem_read` or `dif_otbn_dmem_read`.
+ * @return The result of the operation.
+ */
+static otbn_result_t otbn_check_mem(
+    otbn_t *ctx, const uint8_t *addr, size_t mem_size,
+    dif_result_t (*read_function)(const dif_otbn_t *otbn, uint32_t offset_bytes,
+                                  void *dest, size_t len_bytes)) {
```

Comment:
Would it make sense to use a typedef here? The formatting might be a less exciting that way...